pkg/JuliaInterface/gap/utils.gi

```diff
@@ -4,6 +4,25 @@
 ##
 #############################################################################
 
+BindGlobal( "JuliaSetVal",
+function( name, val )
+  if not IsString( name ) then
+    Error( "JuliaSetVal: <name> must be a string (not the value '", name, "')" );
+  elif Julia.isdefined( Julia.Core,
+         Julia.Symbol( GAPToJulia( "setglobal!" ) ) ) then
+    # Julia 1.9 introduces `setglobal!`.
+    JuliaFunction( "setglobal!" )( Julia.Main,
+        Julia.Symbol( GAPToJulia( name ) ), val );
+  else
+    # `jl_set_global` is available up to Julia 1.8.
+    GAPInfo.JuliaValue:= val;
+    JuliaEvalString( Concatenation(
+        "ccall(:jl_set_global, Cvoid, (Any, Any, Any), Main, Symbol(\"", name,
+        "\"), GAP.Globals.GAPInfo.JuliaValue)" ) );
+    Unbind( GAPInfo.JuliaValue );
+  fi;
+end );
```

Comment:
I would suggest to instead define a Julia helper function which is called here. That then can use `@static if` so that only the version of the code that works in the given Julia version is ever compiled. Something like this ought to work:
```julia
function _setglobal!(M::Module, name::Symbol, val::Any)
  @static if isdefined(Core,:setglobal!)
    Core.setglobal!(M, name, val)
  else
    ccall(:jl_set_global, Cvoid, (Any, Any, Any), M, Symbol(name), val)
    return val
  end
end
```